Fix assist hold to disengagement issue

RetiredBill opened the pull request:
Add small delay in unlatchSignal_BRAKE_uC() after the call to gpio_brakeLights_floatPin(), to allow assist hold to disengage when brake peddle is pressed.

RetiredBill committed:
Fix assist hold issue: not disengaging when brake peddle pressed.

Added 200 uS delay in unlatchSignal_BRAKE_uC() after gpio_brakeLights_floatPin().

RetiredBill committed:
Comment changes only

RetiredBill commented:
Note that I have not looked at either BRAKE_uC or BRAKE_RAW signals with a scope to verify the timing.  I chose the 200 uS value based on the existing comments.

doppelhub commented on muddersMIMA_firmware/brakeLights.cpp:
```diff
@@ -33,17 +33,20 @@ void pulseBrakeLights(void)
 	}
-	
-	gpio_brakeLights_floatPin(); //allows LiControl to check brake status
+    else {
```
Adding 'else' statement changes behavior when the brake pedal is actually pressed.
We should always float the pin at the end of each call, as was done in existing code.
Just add a delay after gpio_brakeLights_floatPin(), as you proposed online.

RetiredBill replied:
Out of an abundance of caution, I was doing a "minimum required change".  The only case I was aware of that needed the delay was after the "gpio_brakeLights_turnOff(); delayMicroseconds(100); gpio_brakeLights_floatPin();" call sequence.  I added the "else {gpio_brakeLights_floatPin();} " to maintain the existing behavior if gpio_getBrakePosition_bool() were to return BRAKE_LIGHTS_ARE_OFF.

It is certainly fine with me to add the delay at the end, so it occurs every time.  You know this code far better than I do! ;)

doppelhub reviewed (changes requested):
Let me know rationale behind new 'else' statement.

RetiredBill committed:
Simpler implementation of delay fix for assist hold issue

Revised comments in unlatchSignal_BRAKE_uC() to more fully explain what is going on.

RetiredBill commented:
I've changed the code as you requested, and done a quick test drive: No surprise, everything works as desired: applying brakes disengages assist hold, brake peddle operates brake lights as expected, as does joystick regen.

I also updated the comments to reflect my understanding of what is going on.  Are my comment changes accurate?
